google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/ManagedCloudSdkService.java

```diff
@@ -16,35 +16,234 @@
+    statusUpdateListeners.remove(listener);
+  }
+
+  public void cancelInstall() {
+    if (runningInstallationJob != null) {
+      runningInstallationJob.cancel(true);
+    }
+  }
+
+  @VisibleForTesting
+  ManagedCloudSdk createManagedSdk() throws UnsupportedOsException {
+    return ManagedCloudSdk.newManagedSdk();
+  }
+
+  @VisibleForTesting
+  void setLogger(Logger logger) {
+    this.logger = logger;
+  }
+
+  @VisibleForTesting
+  void invokeOnApplicationUIThread(Runnable runnable) {
+    ApplicationManager.getApplication().invokeLater(runnable);
+  }
+
+  /**
+   * Checks for Managed Cloud SDK status and creates/installs it if necessary.
+   *
+   * @return Path of the installed managed SDK, null if error occurred while installing.
+   * @throws InterruptedException if Managed Cloud SDK has been interrupted by {@link
+   *     #cancelInstall()}
+   */
+  private Path installManagedSdk() throws InterruptedException {
+    if (managedCloudSdk == null) {
+      try {
+        managedCloudSdk = createManagedSdk();
+      } catch (UnsupportedOsException ex) {
+        logger.error("Unsupported OS for Managed Cloud SDK", ex);
+        updateStatus(SdkStatus.NOT_AVAILABLE);
+        return null;
+      }
+    }
+
+    updateStatus(SdkStatus.INSTALLING);
+
+    Path installedManagedSdkPath;
+    try {
+      installedManagedSdkPath = installSdk();
+    } catch (InterruptedException iex) {
+      updateStatus(SdkStatus.NOT_AVAILABLE);
+      throw iex;
+    } catch (UnsupportedOsException ex) {
+      logger.error("Unsupported OS for Managed Cloud SDK", ex);
+      updateStatus(SdkStatus.NOT_AVAILABLE);
+      return null;
+    } catch (Exception ex) {
+      logger.error("Error while installing managed Cloud SDK", ex);
+      updateStatus(SdkStatus.NOT_AVAILABLE);
+      return null;
+    }
+
+    try {
+      installAppEngineJavaComponent();
+    } catch (InterruptedException iex) {
+      updateStatus(SdkStatus.NOT_AVAILABLE);
+      throw iex;
+    } catch (Exception ex) {
+      logger.error("Error while installing managed Cloud SDK App Engine Java component", ex);
+      updateStatus(SdkStatus.NOT_AVAILABLE);
+      return null;
+    }
+
+    updateStatus(SdkStatus.READY);
+
+    return installedManagedSdkPath;
+  }
+
+  /** Installs core managed SDK if needed and returns its path if successful. */
+  private Path installSdk()
+      throws UnsupportedOsException, CommandExecutionException, InterruptedException, IOException,
+          CommandExitException, SdkInstallerException, UnknownArchiveTypeException {
+    boolean managedSdkInstalled = false;
+    try {
+      managedSdkInstalled = managedCloudSdk.isInstalled();
+    } catch (ManagedSdkVerificationException | ManagedSdkVersionMismatchException ex) {
+      logger.error("Error while checking Cloud SDK status, will attempt to re-install", ex);
+    }
+
+    if (!managedSdkInstalled) {
+      MessageListener sdkInstallListener = logger::trace;
```

Comment:
trace for debugging?

Reply:
SDK installation logs a lot, including all of the CLI style progress (i.e. adding `=` into new line etc.). But still it's probably important info, I agree we can make it `debug`

Reply:
So currently the progress of the installation is simply done in the background and logged to the logger (which the user will probably never look at) correct? What do you think about enhancing this (later) to pipe the progress somewhere visible in the IDE. Something along the lines of what it does when syncing gradle:

<img width="1043" alt="screen shot 2018-02-08 at 11 29 36 am" src="https://user-images.githubusercontent.com/1735744/35984922-68d4e712-0cc3-11e8-8b36-bf2fd7eb0695.png">

Reply:
Yes, that'd be great, I believe Appu just finished new progress bar with meaningful user-visible steps  instead of dumping all of CLI messages, we'll use it to provide step-by-step progress information when we provide UI.